spec/models/survey_response_spec.rb

```diff
@@ -0,0 +1,43 @@
+# == Schema Information
+#
+# Table name: survey_responses
+#
+#  id                         :integer          not null, primary key
+#  perceived_code_confidence  :integer
+#  material_difficulty_rating :integer
+#  notes                      :text
+#  created_at                 :datetime         not null
+#  updated_at                 :datetime         not null
+#
+
+require "rails_helper"
+
+RSpec.describe SurveyResponse, type: :model do
+  it "validates that perceived code confidence is answered" do
+    survey_response = build(:survey_response, perceived_code_confidence: nil)
+    survey_response.valid?
```

Comment:
You don't actually need to call `survey_response.valid?` here because the next line will call it for you :) Same for the other tests in this file.